### PR TITLE
IDEMPIERE-5229 After migrate helper (FHCA-4070)

### PR DESCRIPTION
**Pull request by CarlosRuiz-globalqss**
https://idempiere.atlassian.net/browse/IDEMPIERE-5229

## Checklist:
- [x] My code follows the [code guidelines](https://wiki.idempiere.org/en/Contributing_to_Trunk) of this project
- [x] My code follows the best practices of this project
- [x] I have performed a self-review of my own code
- [x] My code is easy to understand and review. 
- [x] In hard-to-understand areas, I have commented my code.
- [x] My changes generate no new warnings
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added unit tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings
- [x] I have made corresponding changes to the documentation as follows:
- - [x] New feature (non-breaking change which adds functionality): I have created the New Feature page in the project wiki explaining the functionality and how to use it. If relevant, I have committed sample data to the core seed to have usable examples in GardenWorld.
- [x] The changed/added documentation is in the project wiki (not privately-hosted pdf files or links pointing to a company website) and is complete and self-explanatory.

### CODE REVIEW

File: org.adempiere.base/src/org/idempiere/process/DiscardThisVerifyMigration.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	}
+
+	@Override
+	protected String doIt() throws Exception {
+		if (log.isLoggable(Level.INFO))
+			log.info("Process Instance " + p_AD_PInstance_ID);
+
+		int cnt = DB.getSQLValueEx(get_TrxName(), "SELECT COUNT(*) FROM AD_VerifyMigration WHERE AD_PInstance_ID=? AND IsActive='Y' AND IsIgnore='Y'", p_AD_PInstance_ID);
+		if (cnt > 0)
+			return "@Error@ @VM_HasIgnoredRecords@"; // throw new AdempiereException(Msg.getMsg(getCtx(), "VM_HasIgnoredRecords"));
```
Comment (d-ruiz) [inline org.adempiere.base/src/org/idempiere/process/DiscardThisVerifyMigration.java:66]: Why commenting an exception in this part of the code? If the exception is not needed, it is better to delete it completely. 
There's a ticket to just remove this kind of dead code, better to not introduce new one.

File: org.adempiere.base/src/org/idempiere/process/VerifyMigration.java [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+			}
+		}
+	}
+
+	@Override
+	protected String doIt() throws Exception {
+		if (log.isLoggable(Level.INFO))
+			log.info("DateTo" + p_DateTo);
+
+		int cnt = 0;
```
Comment (d-ruiz) [inline org.adempiere.base/src/org/idempiere/process/VerifyMigration.java:86]: couldn't this just be one line ->

int cnt = verifyCustomizationsInChangeLog()  + verifyCustomViewColumns();

File: org.adempiere.base/src/org/idempiere/process/VerifyMigration.java [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+		try (PreparedStatement pstmt = DB.prepareStatement(sql.toString(), get_TrxName())) {
+			ResultSet rs = pstmt.executeQuery();
+			while (rs.next()) {
+				MChangeLog cl = new MChangeLog(getCtx(), rs, get_TrxName());
+				MTable table = MTable.get(cl.getAD_Table_ID());
+				MColumn column = MColumn.get(cl.getAD_Column_ID());
+				PO po = table.getPO(cl.getRecord_ID(), get_TrxName());
+				Object currentValue = po.get_Value(column.getColumnName());
+				String expectedValue = cl.getNewValue();
+				if (   (currentValue == null && expectedValue != null)
```
Comment (d-ruiz) [inline org.adempiere.base/src/org/idempiere/process/VerifyMigration.java:130]: The comment explaining the if should be closer to the if statement, it is far from the code where it is used and quite difficult to understand at first sight.

Comment (CarlosRuiz-globalqss) [reply org.adempiere.base/src/org/idempiere/process/VerifyMigration.java:130]: not sure if you refer to the comment on line 137.

That comment is not about the if - the comment is the value of the message VM_ExpectedValueDiffersFromCurrentValue, it is there to make it clear the variables used in the Msg.getMsg parsing.

File: org.adempiere.base/src/org/idempiere/process/VerifyMigration.java (hunk 2 above)
Comment (CarlosRuiz-globalqss) [reply org.adempiere.base/src/org/idempiere/process/VerifyMigration.java:86]: my plan is to add later more cases - and I think it gives more readability to have the calls in separate lines

File: org.adempiere.base/src/org/idempiere/process/DiscardThisVerifyMigration.java (hunk 1 above)
Comment (CarlosRuiz-globalqss) [reply org.adempiere.base/src/org/idempiere/process/DiscardThisVerifyMigration.java]: fixed - forgot to remove the comment

File: org.adempiere.base/src/org/idempiere/process/VerifyMigration.java (hunk 3 above)
Comment (CarlosRuiz-globalqss) [reply org.adempiere.base/src/org/idempiere/process/VerifyMigration.java:129]: about the comment on the opening of the method - I did a new commit repositioning it

File: org.adempiere.base/src/org/idempiere/process/VerifyMigration.java (hunk 2 above)
Comment (d-ruiz) [reply org.adempiere.base/src/org/idempiere/process/VerifyMigration.java:86]: I think it is less readable and it makes the method longer than it needs to be but as it is a minor thing, so it is not a blocker

**Comment by d-ruiz** [comment]
Looks good and it does not create collateral damage

**Review by d-ruiz — APPROVED** [review]

**Pull request merged**